File: atividade-4/037_rotEsq/037_rotEsq.hs
module Main where

--IN : Um natural n e uma lista ou string S
--OUT: Lista S rotacionada n vezes à esquerda

rotEsq :: Int -> String -> String
rotEsq 0 xs = xs
rotEsq n xs = (rotEsq (n - 1) (tail xs ++ [head xs]))

main = do
  a <- readLn :: IO Int
  b <- getLine
  print $ rotEsq a b

File: atividade-4/037_rotEsq/037_rotEsq.js
//IN : Um natural n e uma lista ou string S
//OUT: Lista S rotacionada n vezes à esquerda

const n = 1;
const w = "asdfg";

const tail = (xs) => xs.slice(1, xs.length);
const head = (xs) => xs[0];

const rotEsq = (n, w) => {
  if (!n) return w;

  return rotEsq(n - 1, tail(w).concat(head(w)));
};

console.log(rotEsq(n, w));
